packages/babel-parser/src/parser/expression.ts

```diff
@@ -3134,16 +3130,14 @@ export default abstract class ExpressionParser extends LValParser {
     return ret;
   }
 
-  // https://github.com/tc39/proposal-js-module-blocks
-  parseModuleExpression(this: Parser): N.ModuleExpression {
-    this.expectPlugin("moduleBlocks");
-    const node = this.startNode<N.ModuleExpression>();
-    this.next(); // eat "module"
+  parseModuleBody(this: Parser, node: Undone<N.ModuleExpression>) {
     if (!this.match(tt.braceL)) {
       this.unexpected(null, tt.braceL);
     }
     // start program node immediately after `{`
     const program = this.startNodeAt<N.Program>(this.state.endLoc);
+    const oldStrict = this.state.strict;
+    this.state.strict = true;
```

Comment:
Note for myself: we also need updating `this.inModule`, checks like `import.meta` and `await` as binding relies on that state.